packages/front-end/components/Experiment/ResultsDownloadButton.tsx

```diff
@@ -0,0 +1,121 @@
+import { ExperimentTableRow, getRisk } from "../../services/experiments";
+
+type UpdatedRow = {
+  metricName?: string;
+};
+
+export default function ResultsDownloadButton({
+  results,
+  experiment,
+}: {
+  results?: ExperimentReportResultDimension;
+  experiment: ExperimentInterfaceStringDates;
+}) {
+  const { getMetricById, ready } = useDefinitions();
+
+  const csvRows = [];
+
+  const rows = useMemo<ExperimentTableRow[]>(() => {
+    if (!results || !results.variations || !ready) return [];
+    return experiment.metrics
+      .map((m) => {
+        const metric = getMetricById(m);
+        return {
+          label: metric?.name,
+          metric,
+          rowClass: metric?.inverse ? "inverse" : "",
+          variations: results.variations.map((v) => {
+            return v.metrics[m];
+          }),
+        };
+      })
+      .filter((row) => row.metric);
+  }, [results, ready]);
+
+  rows.forEach((row) => {
+    row.variations.forEach((variant, index) => {
+      variant.name = experiment.variations[index].name;
+      const { relativeRisk } = getRisk(index, row);
+      variant.relativeRisk = relativeRisk;
+    });
+  });
+
+  rows.forEach((row) => {
+    const updatedRow: UpdatedRow = {};
+    const metric = getMetricById(row.metric.id);
+    updatedRow.metricName = metric.name;
+
+    row.variations.forEach((variation) => {
+      updatedRow[`usersIn${variation.name}`] = variation.users;
+      updatedRow[`countOf${variation.name}`] = variation.value;
+      updatedRow[`conversionRateOf${variation.name}`] = variation.cr;
+      updatedRow[`riskOfChoosing${variation.name}`] = variation.relativeRisk;
+
+      if (variation.chanceToWin) {
+        updatedRow[`chanceToBeatControl`] = variation.chanceToWin;
+      }
+
+      if (variation.expected) {
+        updatedRow[`percentChangeOf${variation.name}`] = variation.expected;
+      }
+    });
+
+    csvRows.push(updatedRow);
+  });
+
+  function generateCsv(data) {
+    const csvRows = [];
+    // Here I need to sort the data[0] keys so 'metricName" is first, with the rest of the keys being in alpha-numeric order
+    const headers = Object.keys(data[0]);
+
+    headers.sort();
+
+    headers.forEach((header, i) => {
+      if (header === "metricName") {
+        headers.splice(i, 1);
+      }
+    });
+
+    headers.unshift("metricName");
+
+    csvRows.push(headers.join(","));
+
+    for (const row of data) {
+      const values = headers.map((header) => {
+        const formattedValues = ("" + row[header]).replace(/"/g, '\\"');
+        return `"${formattedValues}"`;
+      });
+      csvRows.push(values.join(","));
+    }
+    return csvRows.join("\n");
+  }
+
+  const href = useMemo(() => {
+    try {
+      const csv = generateCsv(csvRows);
+      const blob = new Blob([csv], { type: "text/csv" });
+      return window.URL.createObjectURL(blob);
+    } catch (e) {
+      console.error(e);
+      return "";
+    }
+  }, []);
+
+  if (!href) return null;
+
+  return (
```

Comment:
Do I need to add a catch here for if there is no href just to be safe? Given that it's hidden in the menu, the code SHOULD always finish before someone can physically click the button, but maybe try it out.